### ISSUE TITLE
Error：Sorry, this version of Perl (5.018004) is currently not supported by Fink.

### ISSUE THREAD

**Issue by CK-X** [fink/fink-distributions#434]
Mac OS 10.14.5        fink-0.44.1
--------------------------------------------------------------------------

(base) ligengdeMacBook-Pro:~ ligeng$ cd desktop/fink-0.44.1
(base) ligengdeMacBook-Pro:fink-0.44.1 ligeng$ ./bootstrap

Fink must be installed and run with superuser (root) privileges. Fink can
automatically try to become root when it's run from a user account. Since
you're currently running this script as a normal user, the method you choose
will also be used immediately for this script. Available methods:

(1)	Use sudo
(2)	Use su
(3)	None, fink must be run as root

Choose a method: [1] 

Checking package... looks good (fink-0.44.1).
Checking system... i386-apple-darwin18.6.0
This system was not released at the time this Fink release was made. 
Prerelease versions of Mac OS X might work with Fink, but there are no
guarantees.
Distribution: 10.14
Architecture: x86_64
Checking cc... looks good.
Checking make... looks good.
Checking head... looks good.
Verifying that installed Xcode Command Line Tools version is supported...
10.3.0.0.1.1562985497 is OK
Verifying that installed Xcode app version is supported...
Found Xcode version: 10.2.1 is OK.

Please choose the path where Fink should be installed. Note that you will
normally be able to use a binary distribution only if you choose '/sw'. [/sw] 

OK, installing into '/sw'.

Creating directories...
/bin/mkdir -p -m755 /sw
/bin/mkdir -m755 /sw/etc
/bin/mkdir -m755 /sw/etc/alternatives
/bin/mkdir -m755 /sw/etc/apt
/bin/mkdir -m755 /sw/src
/bin/mkdir -m755 /sw/fink
/bin/mkdir -m755 /sw/fink/debs
/bin/mkdir -m755 /sw/var
/bin/mkdir -m755 /sw/var/lib
/bin/mkdir -m755 /sw/var/lib/fink
/bin/mkdir -m755 /sw/fink/10.9-libcxx
/bin/mkdir -m755 /sw/fink/10.9-libcxx/stable
/bin/mkdir -m755 /sw/fink/10.9-libcxx/local
/bin/mkdir -m755 /sw/fink/10.9-libcxx/local/main
/bin/mkdir -m755 /sw/fink/10.9-libcxx/local/main/finkinfo
/bin/mkdir -m755 /sw/fink/10.9-libcxx/local/main/binary-darwin-x86_64
/bin/mkdir -m755 /sw/fink/10.9-libcxx/stable/main
/bin/mkdir -m755 /sw/fink/10.9-libcxx/stable/main/finkinfo
/bin/mkdir -m755 /sw/fink/10.9-libcxx/stable/main/binary-darwin-x86_64
/usr/bin/sed -e "s,@ARV@,local," -e "s,@COMP@,main," -e "s,@ARCH@,x86_64," < Release > /sw/fink/10.9-libcxx/local/main/binary-darwin-x86_64/Release; /usr/bin/touch /sw/fink/10.9-libcxx/local/main/binary-darwin-x86_64/Packages; /usr/bin/gzip /sw/fink/10.9-libcxx/local/main/binary-darwin-x86_64/Packages
/usr/bin/sed -e "s,@ARV@,stable," -e "s,@COMP@,main," -e "s,@ARCH@,x86_64," < Release > /sw/fink/10.9-libcxx/stable/main/binary-darwin-x86_64/Release; /usr/bin/touch /sw/fink/10.9-libcxx/stable/main/binary-darwin-x86_64/Packages; /usr/bin/gzip /sw/fink/10.9-libcxx/stable/main/binary-darwin-x86_64/Packages
Creating fink tarball...
tar -cf /sw/src/fink-0.44.1.tar AUTHORS COPYING INSTALL INSTALL.html MANIFEST MANIFEST.SKIP Makefile NEWS README README.buildsystem README.html README.removing-fink-bld REVISION Release STYLE TODO USAGE USAGE.html VERSION bootstrap bootstrap-phase2.pl compiler_wrapper-10.6.in compiler_wrapper-10.7.in compiler_wrapper-10.9.in dpkg-checkall.sh fink-dpkg-status-cleanup.in fink-instscripts.in fink-scanpackages.in fink-virtual-pkgs.in fink.8.in fink.conf.5.in fink.csh fink.in fink.sh fink.shlibs g++-wrapper.in images/finkDoneFailed.png images/finkDonePassed.png images/finkPackageBuildFailed.png images/finkPackageBuildPassed.png images/finkPackageInstallationFailed.png images/finkPackageInstallationPassed.png images/finkPackageRemovalFailed.png images/finkPackageRemovalPassed.png inject.pl install.sh lockwait.in pathsetup.sh.in perlmod/Fink.pm.in perlmod/Fink/Base.pm perlmod/Fink/Bootstrap.pm perlmod/Fink/CLI.pm perlmod/Fink/Checksum.pm perlmod/Fink/Checksum/MD5.pm perlmod/Fink/Checksum/SHA1.pm perlmod/Fink/Checksum/SHA256.pm perlmod/Fink/Command.pm perlmod/Fink/Config.pm perlmod/Fink/Configure.pm perlmod/Fink/Engine.pm perlmod/Fink/Finally.pm perlmod/Fink/Finally/BuildConflicts.pm perlmod/Fink/Finally/Buildlock.pm perlmod/Fink/FinkVersion.pm.in perlmod/Fink/Mirror.pm perlmod/Fink/NetAccess.pm perlmod/Fink/Notify.pm perlmod/Fink/Notify/Growl.pm perlmod/Fink/Notify/QuickSilver.pm perlmod/Fink/Notify/Say.pm perlmod/Fink/Notify/Syslog.pm perlmod/Fink/Notify/XTerm.pm perlmod/Fink/Package.pm perlmod/Fink/PkgVersion.pm perlmod/Fink/Scanpackages.pm perlmod/Fink/SelfUpdate.pm perlmod/Fink/SelfUpdate/Base.pm perlmod/Fink/SelfUpdate/CVS.pm perlmod/Fink/SelfUpdate/git.pm perlmod/Fink/SelfUpdate/point.pm perlmod/Fink/SelfUpdate/rsync.pm perlmod/Fink/SelfUpdate/svn.pm perlmod/Fink/Services.pm perlmod/Fink/Status.pm perlmod/Fink/SysState.pm perlmod/Fink/Text/DelimMatch.pm perlmod/Fink/Text/ParseWords.pm perlmod/Fink/Tie/IxHash.pm perlmod/Fink/Validation.pm perlmod/Fink/VirtPackage.pm postinstall.pl.in pre-build-test.sh readme.en.html readme.es.html readme.fr.html readme.ja.html readme.pt.html readme.ru.html readme.zh.html setup.sh shlibs.default.in t/00compile.t t/10features/users_groups.t t/10features/volume.t t/Base/initialize.t t/Base/param.t t/Base/param_boolean.t t/CLI/capture.t t/Command/cat.t t/Command/chowname.t t/Command/commands.t t/Command/du_sk.t t/Command/exports.t t/Command/failure.t t/Command/touch.t t/Config/exports.t t/Config/failure.t t/Config/flag.t t/Config/load_save.t t/Config/options.t t/Config/param.t t/Config/verbosity_level.t t/Engine/fetch.t t/Engine/p0.info t/Engine/p1.info t/Engine/p2.info t/Engine/p3.info t/Engine/p4.info t/Finally/buildlocks.t t/Finally/finally.t t/FinkVersion/exports.t t/FinkVersion/version.t t/Mirror/exports.t t/Notify/exports.t t/Package/duplicate_fullname_trees/epoch1/finkinfo/duplicate-fullname.info t/Package/duplicate_fullname_trees/epoch1again/finkinfo/duplicate-fullname.info t/Package/duplicate_fullname_trees/epoch2/finkinfo/duplicate-fullname.info t/Package/duplicate_fullnames.t t/PkgVersion/get_perl.t t/PkgVersion/get_ruby.t t/PkgVersion/get_splitoffs-tree/finkinfo/p1.info t/PkgVersion/get_splitoffs-tree/finkinfo/p2.info t/PkgVersion/get_splitoffs-tree/finkinfo/p3.info t/PkgVersion/get_splitoffs.t t/PkgVersion/mp3-info-pm-1.01-1.info t/PkgVersion/non-consecutive-test.info t/PkgVersion/non-consecutive-test2.info t/PkgVersion/non_consecutive_source.t t/PkgVersion/opengl-rb18.info t/PkgVersion/sed-4.0.5-1.info t/PkgVersion/xml-parser-pm560-2.31-5.info t/SelfUpdate/last_done.t t/SelfUpdate/subclasses.t t/Services/eval_conditional.t t/Services/execute_nonroot_okay.t t/Services/expand_percent.t t/Services/prepare_script.t t/Services/spec2struct.t t/Services/version_cmp.t t/Text/DelimMatch/DelimMatch.t t/Text/ParseWords/ParseWords.t t/Text/ParseWords/taint.t t/Tie/IxHash/each-delete.t t/Tie/IxHash/ixhash.t t/Validation/_filename_versioning_cmp.t t/Validation/exports.t t/basepath/etc/fink.conf t/testmore.pl update-packages/README update/Makefile.in.in update/config.guess update/config.sub update/ltconfig update/ltmain.sh
Copying package description(s)...
/bin/mkdir -p /sw/fink/dists/stable/main/finkinfo/base
/bin/cp 10.9-libcxx/*.info 10.9-libcxx/*.patch /sw/fink/dists/stable/main/finkinfo/base/
echo "bootstrap" > /sw/fink/dists/VERSION
/bin/chmod 644 /sw/fink/dists/VERSION
/bin/mkdir -p -m755 /sw/fink/dists/stable/main/finkinfo/base
Modifying package description...
Use of uninitialized value $Fink::Checksum::SHA256::basepath in concatenation (.) or string at /Users/ligeng/Desktop/fink-0.44.1/perlmod/Fink/Checksum/SHA256.pm line 50.
/bin/chmod 644 /sw/fink/dists/stable/main/finkinfo/base/*.*
Creating initial configuration...

OK, I'll ask you some questions and update the configuration file in
'/sw/etc/fink.conf'.

In what additional directory should Fink look for downloaded tarballs? (use the
absolute pathname only) [] 

Which directory (absolute pathname) should Fink use to build packages? (If you
don't know what this means, it is safe to leave it at its default.) [] 

Fink does not yet support an official set of binary packages for your current
distribution.

=> Setting UseBinaryDist to 'false'

Note: As of the OS X 10.7 distribution, fink no longer has a separate
"unstable" tree. All development and releases happen in the "stable" tree, the
only public tree that exists.

Fink build user settings
Fink can set the UID and GID of its build user dynamically to find unused
values in a safe range.  This is recommended unless you need to have a specific
UID/GID e.g. due to a policy on your network.  Allow Fink to set the UID/GID
dynamically? [Y/n] 
The current ID search range is 600-699.
Using 600

(1)	Quiet (do not show download statistics)
(2)	Low (do not show tarballs being expanded)
(3)	Medium (will show almost everything)
(4)	High (will show everything)

How verbose should Fink be? [2] 

Proxy/Firewall settings
Enter the URL of the HTTP proxy to use, or 'none' for no proxy. The URL should
start with http:// and may contain username, password, and/or port
specifications. Note that this value will be visible to all users on your
computer.
Example, http://username:password@hostname:port
Your proxy:  [none] 
Enter the URL of the FTP proxy to use, or 'none' for no proxy. The URL should
start with http:// and may contain username, password, and/or port
specifications. Note that this value will be visible to all users on your
computer.
Example, http://username:password@hostname:port
Your proxy:  [none] 
Use passive mode FTP transfers (to get through a firewall)? [Y/n] 

Enter the maximum number of simultaneous build jobs. In general, Fink will
build packages faster on systems with multiple CPUs/cores if you allow it to
spawn jobs in parallel. You have 4 active CPUs/cores on your system.
Maximum number of simultaneous build jobs: [4] 

Mirror selection

The Fink team maintains mirrors known as "Master" mirrors, which contain the
sources for all fink packages. You can choose to use these mirrors first, last,
never, or mixed in with regular mirrors. If you don't care, just select the
default.

(1)	Search "Master" source mirrors first.
(2)	Search "Master" source mirrors last.
(3)	Never use "Master" source mirrors.
(4)	Search closest source mirrors first. (combine all mirrors into one set)

What mirror order should fink use when downloading sources? [1] 

Choose a continent:

(1)	Africa
(2)	Asia
(3)	Australia
(4)	Europe
(5)	North America
(6)	South America, Middle America and Caribbean

Your continent? [2] 

Choose a country:

(1)	No selection - display all mirrors on the continent
(2)	Armenia
(3)	Bangladesh
(4)	China
(5)	Georgia
(6)	Hong Kong
(7)	India
(8)	Indonesia
(9)	Iran
(10)	Japan
(11)	Kazakhstan
(12)	Malaysia
(13)	Oman
(14)	Pakistan
(15)	Philippines
(16)	Qatar
(17)	Saudi Arabia
(18)	Singapore
(19)	South Korea
(20)	Taiwan
(21)	Thailand
(22)	Vietnam

Your country? [4] 

Choose a mirror for 'Master Fink mirrors':

(1)	Primary: http://distfiles.master.finkmirrors.net/distfiles/

Mirror for Master Fink mirrors? [1] 

Choose a mirror for 'RSync SelfUpdate':

(1)	Primary: rsync://distfiles.master.finkmirrors.net/finkinfo/

Mirror for RSync SelfUpdate? [1] 

Choose a mirror for 'Apt-Get Repository':

(1)	Primary: http://bindist.finkmirrors.net

Mirror for Apt-Get Repository? [1] 

Choose a mirror for 'GNU software':

(1)	Primary: http://ftpmirror.gnu.org
(2)	China: ftp://mirrors.ustc.edu.cn/gnu
(3)	China: http://mirrors.ustc.edu.cn/gnu

Mirror for GNU software? [1] 

Choose a mirror for 'GNOME':

(1)	Primary: http://ftp.gnome.org/pub/GNOME
(2)	China: https://mirrors.ustc.edu.cn/gnome

Mirror for GNOME? [1] 

Choose a mirror for 'The GIMP':

(1)	Primary: http://download.gimp.org/mirror/pub/gimp

Mirror for The GIMP? [1] 

Choose a mirror for 'Comprehensive TeX Archive Network':

(1)	Primary: ftp://tug.ctan.org/tex-archive
(2)	China: ftp://mirrors.shu.edu.cn/CTAN
(3)	China: ftp://mirrors.ustc.edu.cn/CTAN
(4)	China: http://mirror.lzu.edu.cn/CTAN
(5)	China: http://mirrors.cqu.edu.cn/CTAN
(6)	China: http://mirrors.geekpie.club/CTAN
(7)	China: http://mirrors.shu.edu.cn/CTAN
(8)	China: http://mirrors.tuna.tsinghua.edu.cn/CTAN
(9)	China: http://mirrors.ustc.edu.cn/CTAN

Mirror for Comprehensive TeX Archive Network? [1] 

Choose a mirror for 'Comprehensive Perl Archive Network':

(1)	Primary: ftp://ftp.cpan.org/pub/CPAN
(2)	China: ftp://mirrors.shu.edu.cn/CPAN
(3)	China: ftp://mirrors.ustc.edu.cn/CPAN
(4)	China: http://mirror.lzu.edu.cn/CPAN
(5)	China: http://mirrors.163.com/cpan
(6)	China: http://mirrors.neusoft.edu.cn/cpan
(7)	China: http://mirrors.shu.edu.cn/CPAN
(8)	China: http://mirrors.sohu.com/CPAN
(9)	China: http://mirrors.ustc.edu.cn/CPAN

Mirror for Comprehensive Perl Archive Network? [1] 

Choose a mirror for 'Debian':

(1)	Primary: ftp://ftp.debian.org/debian
(2)	China: http://ftp.cn.debian.org/debian

Mirror for Debian? [1] 

Choose a mirror for 'FreeBSD':

(1)	Primary: http://distcache.FreeBSD.org/ports-distfiles/

Mirror for FreeBSD? [1] 

Choose a mirror for 'KDE':

(1)	Primary: http://download.kde.org/
(2)	China: http://mirrors.ustc.edu.cn/kde

Mirror for KDE? [1] 

Choose a mirror for 'SourceForge':

(1)	Primary: http://downloads.sourceforge.net

Mirror for SourceForge? [1] 

Choose a mirror for 'The Apache Project':

(1)	Primary: http://www.apache.org/dist
(2)	China: http://mirror.bit.edu.cn/apache
(3)	China: http://mirrors.hust.edu.cn/apache
(4)	China: http://mirrors.shu.edu.cn/apache
(5)	China: http://mirrors.tuna.tsinghua.edu.cn/apache

Mirror for The Apache Project? [1] 

Writing updated configuration to '/sw/etc/fink.conf'...
plist is dpkg-bootstrap
Sorry, this version of Perl (5.018004) is currently not supported by Fink.
(base) ligengdeMacBook-Pro:fink-0.44.1 ligeng$

**Comment by nieder** [comment]
Apple changed the version of system-perl 
on the 10.14.5 update (first time they bumped the version in a minor 
release) and that messed things up across the board.

To fix while we get a release out:
1) Update Fink to latest commit that knows about this change:
* Download and extract file
https://github.com/fink/fink/archive/0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b.zip
* `cd fink-0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b`
* run `./inject.pl` to get a fink tool that understands 10.14.5
* After successful completion of the update, check that the 
"Distribution:" field in /sw/etc/fink.conf says 10.14.5.

2) Update your package descriptions for 10.14.5
* Prepare Fink tree for your copy of the updated package descriptions:
`sudo mkdir -p /sw/fink/dists/local/distributions`
`sudo chown -R <USERID>:admin /sw/fink/dists/local/distributions`

* Clone the fink-distribution tree and get the 10.14.5 work-in-progress
branch:
```
cd /sw/fink/dists/local/distributions
git clone https://github.com/fink/fink-distributions.git
cd fink-distributions
git checkout --track origin/dist10.14.5
```

* Tell fink about your new copy of the package descriptions:
Edit /sw/etc/fink.conf and add
`local/distributions/fink-distributions/10.9-libcxx/stable/main` to the
END of the Trees: field. It should at a minimum have (one line):
Trees: stable/main local/main
local/distributions/fink-distributions/10.9-libcxx/stable/main

* run `fink index` to discover the packages available in your new tree. 
To confirm everything worked, `fink list perl51` should show a versioned 
  package for perl5182-core as well as a virtual package for perl5184-core.

**Comment by sydb** [comment]
Thank you, @nieder for the thorough response. I would love to say it fixed my case of the same problem, but in fact I get an error from `./inject.pl`, detailed below. The difference in my case is that I was installing fink-0.44.1 by executing `~/Downloads/InstallFink/Install\ Fink.tool`. Running macOS Mojave 10.14.6 on a MacBook Pro Retina 13-inch mid-2014.
~~~~~
(408) Padfoot ~/Downloads/fink-0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b @ 08:17:02 ->./inject.pl 
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LC_ALL = (unset),
	LC_TIME = "en_DK.UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
This script must be run under sudo, which requires your password.
Password:
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LC_ALL = (unset),
	LC_TIME = "en_DK.UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
The directory '/sw' (guessed) does not contain a Fink installation. Please provide the correct path as a parameter to this
script.
(409) Padfoot ~/Downloads/fink-0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b @ 08:17:12 ->
~~~~~
Note that the `/sw` directory does contain at least a partial installation of fink:
~~~~~
(409) Padfoot ~/Downloads/fink-0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b @ 08:22:57 ->ls -Floagh /sw/
total 0
drwxr-xr-x   6    192B Aug  5 00:44 ./
drwxr-xr-x  30    960B Aug  5 00:44 ../
drwxr-xr-x   5    160B Aug  5 00:45 etc/
drwxr-xr-x   6    192B Aug  5 00:44 fink/
drwxr-xr-x   3     96B Aug  5 00:44 src/
drwxr-xr-x   3     96B Aug  5 00:44 var/
(410) Padfoot ~/Downloads/fink-0c4d2ece1a1fd15a14b0e5f7620a264feee3b40b @ 08:23:22 ->
~~~~~

**Comment by nieder** [comment]
Fink-0.45.0 was released this morning that knows about 10.14.6. If you're starting from scratch, download the [latest version](https://github.com/fink/scripts/releases) of "Install Fink.tool".

Then `sudo rm -r /sw` (BE VERY CAREFUL ABOUT SPACES TO AVOID DELETING YOUR ENTIRE SYSTEM!)

Then run the new `Install\ Fink.tool` to install Fink-0.45.0.

**Comment by nieder** [comment]
Should be fixed.
